test/e2e/singlecluster/pod_test.go

```diff
@@ -522,6 +520,20 @@ var _ = ginkgo.Describe("Pod groups", func() {
 	})
 })
 
+func expectWorkloadFinalized(ctx context.Context, k8sClient client.Client, wlKey client.ObjectKey) {
+	gomega.EventuallyWithOffset(1, func(g gomega.Gomega) error {
+		var wl kueue.Workload
+		err := k8sClient.Get(ctx, wlKey, &wl)
+		if client.IgnoreNotFound(err) != nil {
+			return err
+		}
+		if err != nil || len(wl.Finalizers) == 0 {
+			return nil
+		}
+		return fmt.Errorf("finalizers not yet removed")
```

Comment:
```suggestion
		return fmt.Errorf("Workload %s is not finalized yet", wlKey)
```
nit: I suppose it could be useful for debugging to grep and match different log artifacts

Reply:
Thanks. Fixed.